app/tasks/SyncBooksWithAmazonJob.py

```diff
@@ -0,0 +1,45 @@
+from app.utils.flask_sqlite_queue import Job
+from app import app, queue, db
+from app.models import Book, Genre
+
+from datetime import datetime, timedelta
+
+from amazon.api import AmazonAPI, AsinNotFound
+
+class SyncBooksWithAmazonJob(Job):
+    
+    def __init__(self, repeat = True):
+        Job.__init__(self)
+        self.repeat = repeat
+    
+    def run(self):
+        if self.repeat:
+            self._schedule_next_run()
+        for book in Book.query.all():
+            self._sync_book_with_amazon(book)
+
+    def _schedule_next_run(self):
+        today = datetime.today()
+        tomorrow = today + timedelta(days=1)
+        tomorrow_1_am = datetime(tomorrow.year, tomorrow.month, tomorrow.day, hour = 1, tzinfo = None )
```

Comment:
I would set the hour to 2, so that the amazon job runs each day, one hour after we synched with goodreads :-)